### PR TITLE
MCR 3193 fix pmd issues in mycore mets

### PR DESCRIPTION
**Pull request by Antoniafriedrich**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3193).

### CODE REVIEW

File: mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java [hunk 1]
```diff
@@ -52,10 +53,12 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
...
         while (jsonReader.hasNext()) {
             //CSOFF: InnerAssignment
-            switch (jsonReader.nextName()) {
+            String name = jsonReader.nextName();
+            switch (name) {
                 case "altoFile" -> fileID = jsonReader.nextString();
                 case "begin" -> begin = jsonReader.nextString();
-                case "end" -> end = jsonReader.nextString();
+                case "end"-> end = jsonReader.nextString();
+                default -> throw new JsonParseException(name + " not recognized");
```
Comment (sebhofmann) [inline mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java]: previously unknown names were just ignored.

File: mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java [hunk 2]
```diff
@@ -48,16 +49,19 @@ public MCRMetsLink read(JsonReader jsonReader) throws IOException {
 
         jsonReader.beginObject();
         while (jsonReader.hasNext()) {
-            switch (jsonReader.nextName()) {
+            String name = jsonReader.nextName();
+            switch (name) {
                 case "from" -> ml.setFromString(jsonReader.nextString());
                 case "to" -> ml.setToString(jsonReader.nextString());
+                default -> throw new JsonParseException(name + " is not a valid link type");
```
Comment (sebhofmann) [inline mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java]: previously unknown names were just ignored.

**Review by sebhofmann — CHANGES_REQUESTED** [review]
Some minor issues. Rest looks good.

File: mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java [hunk 3]
```diff
@@ -52,10 +52,13 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
         jsonReader.beginObject();
         while (jsonReader.hasNext()) {
             //CSOFF: InnerAssignment
-            switch (jsonReader.nextName()) {
-                case "altoFile" -> fileID = jsonReader.nextString();
-                case "begin" -> begin = jsonReader.nextString();
-                case "end" -> end = jsonReader.nextString();
```
Comment (rsteph-de) [inline mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:58]: We want to keep the switch and add a default branch

File: mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java [hunk 4]
```diff
@@ -48,16 +48,19 @@ public MCRMetsLink read(JsonReader jsonReader) throws IOException {
 
         jsonReader.beginObject();
         while (jsonReader.hasNext()) {
-            switch (jsonReader.nextName()) {
-                case "from" -> ml.setFromString(jsonReader.nextString());
-                case "to" -> ml.setToString(jsonReader.nextString());
```
Comment (rsteph-de) [inline mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java:53]: We want to keep the switch and add a default branch

**Comment by rsteph-de** [comment]
2024-10-15T07:13:53.4502745Z 07:13:53,449 [INFO] There are 3 errors reported by Checkstyle 10.12.7 with /home/runner/work/mycore/mycore/checkstyle.xml ruleset.
2024-10-15T07:13:53.4527722Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[55,43] (coding) InnerAssignment: Inner assignments should be avoided.
2024-10-15T07:13:53.4531927Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[56,39] (coding) InnerAssignment: Inner assignments should be avoided.
2024-10-15T07:13:53.4536340Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[57,35] (coding) InnerAssignment: Inner assignments should be avoided.

File: mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java [hunk 5]
```diff
@@ -51,13 +51,14 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
 
         jsonReader.beginObject();
         while (jsonReader.hasNext()) {
-            //CSOFF: InnerAssignment
```
Comment (rsteph-de) [inline mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java]: the 2 comments need to be added to disable checkstyle here

**Review by rsteph-de — APPROVED** [review]
Looks good for me now

**Pull request merged**